src/main/java/com/google/javascript/gents/TypeScriptGenerator.java

```diff
@@ -176,26 +183,42 @@ private void generateTypeScript() {
 
     new StyleFixPass(compiler, comments).process(externRoot, srcRoot);
 
+    new AstValidator(
+            compiler,
+            new AstValidator.ViolationHandler() {
+              @Override
+              public void handleViolation(String message, Node n) {
+                compiler.report(JSError.make(n, CLUTZ_MALFORMED_AST, message));
+              }
+            })
+        .process(externRoot, srcRoot);
+
     // We only use the source root as the extern root is ignored for codegen
     for (Node file : srcRoot.children()) {
-      String filepath = pathUtil.getFilePathWithoutExtension(file.getSourceFileName());
-      CodeGeneratorFactory factory = new CodeGeneratorFactory() {
-        @Override
-        public CodeGenerator getCodeGenerator(Format outputFormat, CodeConsumer cc) {
-          return new GentsCodeGenerator(cc, compilerOpts, comments, opts.externsMap);
-        }
-      };
-
-      String tsCode = new CodePrinter.Builder(file)
-          .setCompilerOptions(opts.getCompilerOptions())
-          .setTypeRegistry(compiler.getTypeRegistry())
-          .setCodeGeneratorFactory(factory)
-          .setPrettyPrint(true)
-          .setLineBreak(true)
-          .setOutputTypes(true)
-          .build();
-
-      sourceFileMap.put(filepath, tryClangFormat(tsCode));
+      try {
+        String filepath = pathUtil.getFilePathWithoutExtension(file.getSourceFileName());
+        CodeGeneratorFactory factory =
+            new CodeGeneratorFactory() {
+              @Override
+              public CodeGenerator getCodeGenerator(Format outputFormat, CodeConsumer cc) {
+                return new GentsCodeGenerator(cc, compilerOpts, comments, opts.externsMap);
+              }
+            };
+
+        String tsCode =
+            new CodePrinter.Builder(file)
+                .setCompilerOptions(opts.getCompilerOptions())
+                .setTypeRegistry(compiler.getTypeRegistry())
+                .setCodeGeneratorFactory(factory)
+                .setPrettyPrint(true)
+                .setLineBreak(true)
+                .setOutputTypes(true)
+                .build();
+
+        sourceFileMap.put(filepath, tryClangFormat(tsCode));
+      } catch (Throwable t) {
+        System.err.println("Failed while converting " + file.getSourceFileName());
```

Comment:
Yeah, but it's even better when I rethrow the exception so we don't just silently fail (`o_O`)